app/models/translation.rb

```diff
@@ -18,6 +32,13 @@ def time_updated
     self.time_last_updated
   end
 
+  def update_status_to_completed
+    unless self.status == @@COMPLETE_WITH_PRIORITY_STATUS
+      self.status = @@COMPLETE_WITH_PRIORITY_STATUS if self.video.starred else @@COMPLETE_STATUS
+      self.save
+    end
+  end
+
   def complete?
     File.exist?(srt_path)
```

Comment:
With the introduction of your status fields, we can now make this method less jank.